src/lib/Components/Bidding/Elements/types.d.ts

```diff
@@ -0,0 +1,210 @@
+
+// border
+
+export type BorderRadiusValue = number
+
+export interface BorderRadiusProps {
+  borderRadius?: BorderRadiusValue
+}
+
+export function borderRadius(...args: any[]): any
+
+export type BorderColorValue = string
+
+export interface BorderColorProps {
+  borderColor?: BorderColorValue
+}
+
+export function borderColor(...args: any[]): any
+
+export type BorderWidthValue = number
+
+export interface BorderWidthProps {
+  borderWidth?: BorderWidthValue
+  borderTop?: boolean
+  borderRight?: boolean
+  borderBottom?: boolean
+  borderLeft?: boolean
+}
+
+export function borderWidth(...args: any[]): any
+
+// flex
+export type AlignItemsValue =
+  | "normal"
+  | "stretch"
+  | "center"
+  | "start"
+  | "end"
+  | "flex-start"
+  | "flex-end"
+  | "self-start"
+  | "self-end"
+  | "left"
+  | "right"
+  | "baseline"
+  | "first baseline"
+  | "last baseline"
+  | "safe center"
+  | "unsafe center"
+export type ResponsiveAlignItemsValue = ResponsiveValue<AlignItemsValue>
+
+export interface AlignItemsProps {
+  alignItems?: ResponsiveAlignItemsValue
+}
+
+export function alignItems(...args: any[]): any
+
+export type JustifyContentValue =
+  | "center"
+  | "start"
+  | "end"
+  | "flex-start"
+  | "flex-end"
+  | "left"
+  | "right"
+  | "baseline"
+  | "first baseline"
+  | "last baseline"
+  | "space-between"
+  | "space-around"
+  | "space-evenly"
+  | "stretch"
+  | "safe center"
+  | "unsafe center"
+export type ResponsiveJustifyContentValue = ResponsiveValue<JustifyContentValue>
+
+export interface JustifyContentProps {
+  justify?: ResponsiveJustifyContentValue
+}
+
+export function justifyContent(...args: any[]): any
+
+export interface FlexWrapProps {
+  wrap?: boolean
+}
+
+export function flexWrap(...args: any[]): any
+
+export type FlexDirectionValue = "row" | "row-reverse" | "column" | "column-reverse"
+export type ResponsiveFlexDirectionValue = ResponsiveValue<FlexDirectionValue>
+
+export interface FlexDirectionProps {
+  flexDirection?: ResponsiveFlexDirectionValue
+}
+
+export function flexDirection(...args: any[]): any
+
+export type FlexValue = string
+export type ResponsiveFlexValue = ResponsiveValue<FlexValue>
+
+export interface FlexProps {
+  flex?: ResponsiveFlexValue
+}
+
+export function flex(...args: any[]): any
+
+export type AlignSelfValue =
+  | "auto"
+  | "normal"
+  | "center"
+  | "start"
+  | "end"
+  | "self-start"
+  | "self-end"
+  | "flex-start"
+  | "flex-end"
+  | "left"
+  | "right"
+  | "baseline"
+  | "first baseline"
+  | "last baseline"
+  | "stretch"
+  | "safe center"
+  | "unsafe center"
+export type ResponsiveAlignSelfValue = ResponsiveValue<AlignSelfValue>
+
+export interface AlignSelfProps {
+  alignSelf?: ResponsiveAlignSelfValue
+}
+
+export function alignSelf(...args: any[]): any
+
+export interface FlexboxProps
+  extends AlignItemsProps,
+    JustifyContentProps,
+    FlexWrapProps,
+    FlexDirectionProps,
+    FlexProps,
+    AlignSelfProps {}
```

Comment:
This `types.d` file will be very useful elsewhere. One comment / question: I wonder if some of these type values might be available via the `styled-components` or `react` packages -- things like flex and so on, so that they don't need to be recreated?

Reply:
Currently, `styled-system` [doesn't have type definitions](https://github.com/jxnblk/styled-system/issues/71) in the repo or in DefinitelyTyped. I took most of the definitions in [this gist](https://gist.github.com/maxdeviant/cfed775121829eb02283347083915383), but even this is missing a lot of definitions.

In order to get the best possible suggestions in the `styled-system` world, all CSS properties would need to be mapped to React's prop types. I tried to use [`dts-gen`](https://github.com/Microsoft/dts-gen) to generate a stub as a starter, but all it generated was basically just +3000 lines of `any`s, and we need something smarter. I also looked at the source code of `styled-system`, and it's actually way shorter than I'd expected... I'm sure there must be a better way to auto-generate these types and keep them in DefinitelyTyped, but I wanted to get this in a mergeable state.

Reply:
@yuki24  Let me know if you want to discuss automated options to potentially generate these.